sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -176,6 +176,30 @@ def _from_generated(cls, definition):
         )
 
 
+class KeyVaultRoleDefinitionProperties(object):
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        self.principal_id = kwargs.get("principal_id")
+        self.role_definition_id = kwargs.get("role_definition_id")
+        self.scope = kwargs.get("scope")
+
+    def __repr__(self):
+        # type: () -> str
+        return "KeyVaultRoleAssignmentProperties(principal_id={}, role_definition_id={}, scope={})".format(
+            self.principal_id, self.role_definition_id, self.scope
+        )[:1024]
+
+    @classmethod
+    def _from_generated(cls, role_assignment_properties):
+        # the generated RoleAssignmentProperties and RoleAssignmentPropertiesWithScope
+        # models differ only in that the latter has a "scope" attribute
+        return cls(
+            principal_id=role_assignment_properties.principal_id,
+            role_definition_id=role_assignment_properties.role_definition_id,
+            scope=getattr(role_assignment_properties, "scope", None),
+        )
```

Comment:
Discarding the permissions (assuming this method expects RoleDefinitionProperties)?

Reply:
I haven't implemented the body of this method yet so most of this is likely a hot mess 🥳